backend-rust/src/indexer.rs

```diff
@@ -1917,12 +1917,87 @@ impl PreparedAccountDelegationEvent {
                 .execute(tx.as_mut())
                 .await?;
             }
-            PreparedAccountDelegationEvent::NoOperation {} => (),
+            PreparedAccountDelegationEvent::RemoveBaker(baker_removed) => {
+                baker_removed.save(tx).await?;
+            }
         }
         Ok(())
     }
 }
 
+/// Represents the event of a baker being removed, resulting in the delegators
+/// targeting the pool are moved to the passive pool.
+struct BakerRemoved {
+    remove_baker:    RemoveBaker,
+    move_delegators: MovePoolDelegatorsToPassivePool,
+}
+impl BakerRemoved {
+    fn prepare(baker_id: &sdk_types::BakerId) -> anyhow::Result<Self> {
+        Ok(Self {
+            remove_baker:    RemoveBaker::prepare(baker_id)?,
+            move_delegators: MovePoolDelegatorsToPassivePool::prepare(baker_id)?,
+        })
+    }
+
+    async fn save(
+        &self,
+        tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
+    ) -> anyhow::Result<()> {
+        self.remove_baker.save(tx).await?;
```

Comment:
Not that it matters since it is a transaction, but would probably be more natural to reverse the order of these two.